src/containers/GlobalMenu.js

```diff
@@ -0,0 +1,57 @@
+import { currentUserQuery, signInMutation } from '../apollo/queries'
+import { compose, graphql, withApollo } from 'react-apollo'
+import { GlobalMenu } from '../components'
+import { userValidations } from '../helpers/validations'
+import _ from 'lodash'
+
+const signOut = (client) => () => {
+  localStorage.removeItem('auth-token')
+  return client.resetStore()
+}
+const signIn = (client, token) => {
+  localStorage.setItem('auth-token', token)
+  client.resetStore()
+}
+
+const GlobalMenuWithData = compose(
+  withApollo,
+  graphql(currentUserQuery, {
+    props: (props) => {
+      if (
+        !_.get(props, 'data.error.networkError') &&
+        (props.data.error || (!props.data.me && !props.data.loading))
+      ) {
+        // Invalid user token
+        signOut(props.ownProps.client)()
+      }
+      return {
+        ...props.ownProps,
+        loading: _.get(props, 'data.loading'),
+        error: _.get(props, 'data.error'),
+        me: _.get(props, 'data.me')
```

Comment:
I think I initially used it because I checked for nested data and that way we don't have to check each nesting. Sort of like `props&.data&.me` in Ruby. But I don't think that's necessary here anymore.